frontend/js/src/settings/music-services/details/MusicServices.tsx

```diff
@@ -92,6 +101,66 @@ export default function MusicServices() {
     }
   };
 
+  const handleAppleMusicPermissionChange = async (
+    serviceName: string,
+    action: string
+  ) => {
+    try {
+      await loadAppleMusicKit();
+      const musicKitInstance = await setupAppleMusicKit(
+        appleAuth?.developer_token
+      );
+      // Delete or recreate the user in the database for this external service
+      const response = await fetch(
+        `/settings/music-services/apple/disconnect/`,
+        {
+          method: "POST",
+          body: JSON.stringify({ action }),
+          headers: {
+            "Content-Type": "application/json",
+          },
+        }
+      );
+      if (action === "disable") {
+        await musicKitInstance.unauthorize();
+        (appleAuth as AppleMusicUser).music_user_token = undefined;
+        toast.success(
+          <ToastMsg
+            title="Success"
+            message="Apple Music integration has been disabled."
+          />
+        );
+      } else {
+        // authorizeWithAppleMusic also sends the token to the server
+        const newToken = await authorizeWithAppleMusic(musicKitInstance);
+        if (newToken) {
+          // We know appleAuth is not undefined because we needed the developer_token
+          // it contains in order to authorize the user successfully
+          (appleAuth as AppleMusicUser).music_user_token = newToken;
+        }
+        toast.success(
+          <ToastMsg
+            title="Success"
+            message="You are now logged in to Apple Music."
+          />
+        );
+      }
+
+      setPermissions((prevState) => ({
+        ...prevState,
+        appleMusic: action,
+      }));
```

Comment:
You also need to update the context similar to how it is done with other external services

Reply:
That's already dona above on lines 126 and 139 (depending on the action)